introspection-engine/connectors/sql-introspection-connector/src/introspection_helpers.rs

```diff
@@ -218,10 +222,10 @@ pub(crate) fn calculate_backrelation_field(
 
             // unique or id
             let other_is_unique = table
-                .indices
-                .iter()
-                .any(|i| columns_match(&i.columns, &relation_info.fields) && i.tpe == IndexType::Unique)
-                || columns_match(&table.primary_key_columns(), &relation_info.fields);
+                .indexes()
+                .filter(|idx| idx.index_type().is_unique())
+                .any(|i| i.column_names_match(&relation_info.fields))
+                || columns_match(&table.primary_key_column_names().unwrap_or(&[]), &relation_info.fields);
```

Comment:
`columns_match` explicitly compares the columns disregarding the order — I don't understand why, shouldn't they be compared in order?